src/components/ErrorModal.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { Box, Typography, Button, Modal } from '@mui/material';
+
+interface ErrorModalProps {
+  isOpen: boolean;
+  description: string | null;
+  onClose: () => void;
+}
+
+const ErrorModal: React.FC<ErrorModalProps> = ({ isOpen, description, onClose }) => {
+  return (
+    <Modal id="error-modal" open={isOpen} onClose={onClose}>
+      <Box
+        sx={{
+          position: 'absolute',
+          top: '50%',
+          left: '50%',
+          transform: 'translate(-50%, -50%)',
+          width: 400,
+          backgroundColor: 'background.paper',
+          boxShadow: 24,
+          p: 4,
```

Comment:
I'm seeing a few places where you are relying on default units in css. This is a bad practice because not all browsers handle the default the same - older tend to fall back on pixels, newer on rem.